quinn/src/send_stream.rs

```diff
@@ -54,7 +55,50 @@ where
         WriteAll { stream: self, buf }
     }
 
+    /// Write chunks to the stream
+    ///
+    /// Yields the number of bytes and chunks written on success.
+    /// Congestion and flow control may cause this to be shorter than `buf.len()`,
+    /// indicating that only a prefix of `bufs` was written
+    pub fn write_chunks<'a>(&'a mut self, bufs: &'a mut [Bytes]) -> WriteChunks<'a, S> {
+        WriteChunks { stream: self, bufs }
+    }
+
+    /// Convenience method to write a single chunk to the stream
+    pub fn write_chunk(&mut self, buf: Bytes) -> WriteChunk<'_, S> {
+        WriteChunk {
+            stream: self,
+            buf: [buf],
+        }
+    }
+
+    /// Convenience method to write an entire list of chunks to the stream
+    pub fn write_all_chunks<'a>(&'a mut self, bufs: &'a mut [Bytes]) -> WriteAllChunks<'a, S> {
+        WriteAllChunks {
+            stream: self,
+            bufs,
+            offset: 0,
+        }
+    }
+
     fn poll_write(&mut self, cx: &mut Context, buf: &[u8]) -> Poll<Result<usize, WriteError>> {
+        let stream_id = self.stream;
+        self.execute_poll(cx, |conn| conn.write(stream_id, buf))
+    }
+
+    fn poll_write_chunks(
+        &mut self,
+        cx: &mut Context,
+        bufs: &mut [Bytes],
+    ) -> Poll<Result<Written, WriteError>> {
+        let stream_id = self.stream;
+        self.execute_poll(cx, |conn| conn.write_chunks(stream_id, bufs))
+    }
```

Comment:
Would it be more concise to inline these now-trivial methods into the respective `Future::poll` impls?

Reply:
It has it's pro's and con's. Of course they are unnecessary, but they might make it easier to read the poll methods since the readers of those now don't have to understand how the closure thingy works. I don't have any strong opinion.

Reply:
I updated it now, since the connection refactoring which made `proto::SendStream` public made the closure a bit easier to understand.